### PR TITLE
 perfs: optimize script.decomplie return type

### PR DESCRIPTION
**Pull request by jasonandjay**
Optimize script.decomplie return type to Stack 
1. avoid type error. for example: null.map https://github.com/bitcoinjs/bitcoinjs-lib/issues/2096
2. avoid unnecessary type assertion
3. simplify the judgment of returned results

### CODE REVIEW

**Comment by junderw** [comment]
There is a semantic difference between `[]` and `null`

`null` indicates "the decompilation failed / there is no valid decompilation that exists for this Buffer" without need for throwing an exception.

`[]` means "we decompiled it and it was an empty script."

By collapsing this, `[]` can now mean two things.

This is a breaking change.

I agree that the bug in the other issue should be fixed, but this is not the way we should fix it.

Thanks for the PR though.

**Comment by jasonandjay** [comment]
> There is a semantic difference between `[]` and `null`
> 
> `null` indicates "the decompilation failed / there is no valid decompilation that exists for this Buffer" without need for throwing an exception.
> 
> `[]` means "we decompiled it and it was an empty script."
> 
> By collapsing this, `[]` can now mean two things.
> 
> This is a breaking change.
> 
> I agree that the bug in the other issue should be fixed, but this is not the way we should fix it.
> 
> Thanks for the PR though.

I agree

How about throw error replace return null ?

**Comment by junderw** [comment]
I'd rather not have to thing about the knock on effects.

A simple bug fix is easier to approve.

**Comment by jasonandjay** [comment]
@junderw how about this?
just throw an error from toASM when decompile failed
Minimal impact on existing logic

File: src/script.js [hunk 1]
```diff
@@ -158,6 +158,9 @@ function toASM(chunks) {
   if (chunksIsBuffer(chunks)) {
     chunks = decompile(chunks);
   }
+  if (!chunks) {
+    throw new Error('convert invalid chunks to ASM');
```
Comment (junderw) [inline src/script.js:162]: This should be `'Could not convert invalid chunks to ASM'` instead.

**Comment by jasonandjay** [comment]
Please help reRun actions
it`s failed for netowrk issue, reRun failed works will repair

**Pull request merged**